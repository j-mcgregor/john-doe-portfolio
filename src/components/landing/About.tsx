import React, { useState } from 'react'
import { AboutProps } from '../../types/interfaces'

const About: React.FC<AboutProps> = ({ about, experience }) => {
    const [showAbout, setShowAbout] = useState(true)

    const handleToggleExperience = () => setShowAbout(false)
    const handleToggleAbout = () => setShowAbout(true)

    return (
        <section className="container-fluid py4 py2-md about">
            <div className="container">
                <div className="tabs">
                    <div className="tab" onClick={handleToggleAbout}>
                        About
                    </div>
                    <div className="tab" onClick={handleToggleExperience}>
                        Experience
                    </div>
                </div>
                <div className="row">
                    <section className="content col-lg-6 col-lg-offset-3 col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1 py4 py2-md text-justify text-left-sm">
                        {showAbout ? about : experience}
                    </section>
                </div>
            </div>
        </section>
    )
}

export default About
