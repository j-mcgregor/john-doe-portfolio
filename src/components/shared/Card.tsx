import React from 'react'
import RichText from '../../utils/RichTextCustom'
import { PRISMIC_ExperienceType } from '../../types/interfaces/prismic'

export interface CardProps {
    item: PRISMIC_ExperienceType
}

const Card: React.FC<CardProps> = ({ item }) => {
    return (
        <div className="mb3">
            <div className="text-center">
                <h5 className="m0">
                    {item.job_title} @ {item.company}
                </h5>
                <div className="my1">{item.city}</div>
                <small>
                    {item.date_from} to{' '}
                    {item.present ? 'present' : item.date_to}
                </small>
            </div>
            <RichText render={item.description} />
        </div>
    )
}

export default Card
