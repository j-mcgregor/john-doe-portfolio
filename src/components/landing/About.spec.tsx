import React from 'react'
import { render } from '@testing-library/react'
import About from './About'
import RichText from '../../utils/RichTextCustom'
import {
    PRISMIC_RichTextType,
    PRISMIC_ExperienceType,
} from '../../types/interfaces/prismic'
import Card from '../shared/Card'

describe('About', () => {
    // test about section renders correctly
    // test experiences section renders correctly
    // test buttons work
    let paragraph: PRISMIC_RichTextType[]
    let experience: PRISMIC_ExperienceType[]

    beforeEach(() => {
        paragraph = [
            {
                text: 'lorem ipsum',
                type: 'paragraph',
                spans: [],
            },
        ]
        experience = [
            {
                city: 'Oz',
                company: 'Big Wizard Company',
                date_from: '2000-01-01',
                date_to: '2010-12-31',
                description: [
                    {
                        text: 'operated the curtain pulley',
                        type: 'paragraph',
                        spans: [],
                    },
                ],
                job_title: 'Curtain Puller',
            },
        ]
    })
    it('should render the about section by default', () => {
        // assemble
        const about = <RichText render={paragraph} />
        const cards = experience.map(e => <Card key="hi" item={e} />)
        const result = render(<About about={about} experience={cards} />)

        // assert
        expect(result.container).toMatchSnapshot()
        expect(result.getByText(/lorem ipsum/)).toBeDefined()
    })

    it('should display the experience section when tab is clicked', () => {
        // assemble
        const about = <RichText render={paragraph} />
        const cards = experience.map(e => <Card key="hi" item={e} />)
        const result = render(<About about={about} experience={cards} />)

        // act
        result.getByText('Experience').click()

        // assert
        expect(result.container).toMatchSnapshot()
        expect(result.getByText(/Big Wizard Company/)).toBeDefined()
    })
    it('should display the experience section when tab is clicked', () => {
        // assemble
        const about = <RichText render={paragraph} />
        const cards = experience.map(e => <Card key="hi" item={e} />)
        const result = render(<About about={about} experience={cards} />)

        // act
        result.getByText('About').click()

        // assert
        expect(result.container).toMatchSnapshot()
        expect(result.getByText(/lorem ipsum/)).toBeDefined()
    })
})
