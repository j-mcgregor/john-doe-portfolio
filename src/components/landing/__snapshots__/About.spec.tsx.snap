// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should display the experience section when tab is clicked 1`] = `
<div>
  <section
    class="container-fluid py4 py2-md about"
  >
    <div
      class="container"
    >
      <div
        class="tabs"
      >
        <div
          class="tab"
        >
          About
        </div>
        <div
          class="tab"
        >
          Experience
        </div>
      </div>
      <div
        class="row"
      >
        <section
          class="content col-lg-6 col-lg-offset-3 col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1 py4 py2-md text-justify text-left-sm"
        >
          <div
            class="mb3"
          >
            <div
              class="text-center"
            >
              <h5
                class="m0"
              >
                Curtain Puller
                 @ 
                Big Wizard Company
              </h5>
              <div
                class="my1"
              >
                Oz
              </div>
              <small>
                2000-01-01
                 to
                 
                2010-12-31
              </small>
            </div>
            <p>
              operated the curtain pulley
            </p>
          </div>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`About should display the experience section when tab is clicked 2`] = `
<div>
  <section
    class="container-fluid py4 py2-md about"
  >
    <div
      class="container"
    >
      <div
        class="tabs"
      >
        <div
          class="tab"
        >
          About
        </div>
        <div
          class="tab"
        >
          Experience
        </div>
      </div>
      <div
        class="row"
      >
        <section
          class="content col-lg-6 col-lg-offset-3 col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1 py4 py2-md text-justify text-left-sm"
        >
          <p>
            lorem ipsum
          </p>
        </section>
      </div>
    </div>
  </section>
</div>
`;

exports[`About should render the about section by default 1`] = `
<div>
  <section
    class="container-fluid py4 py2-md about"
  >
    <div
      class="container"
    >
      <div
        class="tabs"
      >
        <div
          class="tab"
        >
          About
        </div>
        <div
          class="tab"
        >
          Experience
        </div>
      </div>
      <div
        class="row"
      >
        <section
          class="content col-lg-6 col-lg-offset-3 col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1 py4 py2-md text-justify text-left-sm"
        >
          <p>
            lorem ipsum
          </p>
        </section>
      </div>
    </div>
  </section>
</div>
`;
